src/query/src/range_select/plan.rs

```diff
@@ -117,32 +120,99 @@ impl Fill {
                             data[i] = data[i - 1].clone()
                         }
                     }
-                    Fill::Linear => {
-                        if 0 < i && i < len - 1 {
-                            match (&data[i - 1], &data[i + 1]) {
-                                (ScalarValue::Float64(Some(a)), ScalarValue::Float64(Some(b))) => {
-                                    data[i] = ScalarValue::Float64(Some((a + b) / 2.0));
-                                }
-                                (ScalarValue::Float32(Some(a)), ScalarValue::Float32(Some(b))) => {
-                                    data[i] = ScalarValue::Float32(Some((a + b) / 2.0));
-                                }
-                                (a, b) => {
-                                    if !a.is_null() && !b.is_null() {
-                                        return Err(DataFusionError::Execution(
-                                            "RangePlan: Apply Fill LINEAR strategy on Non-floating type".to_string()));
-                                    } else {
-                                        continue;
-                                    }
-                                }
-                            }
-                        }
-                    }
+                    Fill::Linear => unreachable!(),
```

Comment:
Add some document about this `unreachable!()`

Reply:
done